e2e/ls_test.go

```diff
@@ -391,3 +394,393 @@ func TestListS3ObjectsWithDashH(t *testing.T) {
 		1: match(`^ 264.0K testfile2.txt$`),
 	}, trimMatch(dateRe), alignment(true))
 }
+
+// ls --exclude "*.txt" s3://bucket/*
+func TestListS3ObjectsWithExcludeFilter(t *testing.T) {
+	t.Parallel()
+
+	bucket := s3BucketFromTestName(t)
+
+	const (
+		content        = "content"
+		excludePattern = "*.txt"
+	)
+
+	filenames := []string{
+		"file.txt",
+		"file.py",
+		"hasan.txt",
+		"a/try.txt",
+		"a/try.py",
+		"a/file.c",
+		"file2.txt",
+	}
+
+	s3client, s5cmd, cleanup := setup(t)
+	defer cleanup()
+
+	createBucket(t, s3client, bucket)
+
+	for _, filename := range filenames {
+		putFile(t, s3client, bucket, filename, content)
+	}
+
+	cmd := s5cmd("ls", "--exclude", excludePattern, "s3://"+bucket+"/*")
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Success)
+
+	assertLines(t, result.Stdout(), map[int]compareFunc{
+		0: match(`a/file.c`),
+		1: match(`a/try.py`),
+		2: match(`file.py`),
+	}, trimMatch(dateRe), alignment(true))
+}
+
+// ls --exclude "*.txt" s3://bucket/*
+func TestListS3ObjectsWithFileExtensionsExcludeFilter(t *testing.T) {
+	t.Parallel()
+
+	bucket := s3BucketFromTestName(t)
+
+	const (
+		content        = "content"
+		excludePattern = "*.txt"
+	)
+
+	filenames := []string{
+		"file.txt",
+		"file.py",
+		"hasan.txt",
+		"a/try.txt",
+		"a/try.py",
+		"a/file.c",
+		"file2.txt",
+		"file2.txt.extension", // this should not be excluded.
+	}
+
+	s3client, s5cmd, cleanup := setup(t)
+	defer cleanup()
+
+	createBucket(t, s3client, bucket)
+
+	for _, filename := range filenames {
+		putFile(t, s3client, bucket, filename, content)
+	}
+
+	cmd := s5cmd("ls", "--exclude", excludePattern, "s3://"+bucket+"/*")
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Success)
+
+	assertLines(t, result.Stdout(), map[int]compareFunc{
+		0: match(`a/file.c`),
+		1: match(`a/try.py`),
+		2: match(`file.py`),
+		3: match(`file2.txt.extension`),
+	}, trimMatch(dateRe), alignment(true))
+}
+
+// ls --exclude ".txt" --exclude ".py" s3://bucket
+func TestListS3ObjectsWithExcludeFilters(t *testing.T) {
+	t.Parallel()
+
+	bucket := s3BucketFromTestName(t)
+
+	const (
+		content         = "content"
+		excludePattern1 = "*.txt"
+		excludePattern2 = "*.py"
+	)
+
+	filenames := []string{
+		"file.txt",
+		"file.py",
+		"hasan.txt",
+		"a/try.txt",
+		"a/try.py",
+		"a/file.c",
+		"file2.txt",
+	}
+
+	s3client, s5cmd, cleanup := setup(t)
+	defer cleanup()
+
+	createBucket(t, s3client, bucket)
+
+	for _, filename := range filenames {
+		putFile(t, s3client, bucket, filename, content)
+	}
+
+	cmd := s5cmd("ls", "--exclude", excludePattern1, "--exclude", excludePattern2, "s3://"+bucket)
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Success)
+
+	assertLines(t, result.Stdout(), map[int]compareFunc{
+		0: match(`DIR a/`),
+	}, trimMatch(dateRe), alignment(true))
+}
+
+// ls --exclude "" s3://bucket
+func TestListS3ObjectsWithExcludeFilterEmpty(t *testing.T) {
```

Comment:
```suggestion
func TestListS3ObjectsWithEmptyExcludeFilter(t *testing.T) {
```

Reply:
changed to `TestListS3ObjectsWithEmptyExcludeFilter`.